### PR TITLE
Add ciao-webui support to ciao-down and Single VM

### PR DESCRIPTION
**Pull request by markdryan**
This PR adds ciao-webui support to ciao-down and Single VM.   In short ciao-down installs and configures node and downloads the ciao-webui repo into a directory shared with the host.  Single VM creates a configuration file that contains all the information the webui needs to talk to the Single VM ciao instance.

We also fix a bug which causes a launcher crash when container creation fails due to a network error.

### CODE REVIEW

**Comment by markdryan** [comment]
@jorgevgut I need your help with this one.  The ciao-webui is working okay but I cannot build the latest version of the code inside single VM, e.g., make install-dev errors out.  It seems to be complaining about proxy issues but I have set the proxies using npm config.

Perhaps there are other npm configs I need to set, like the prefix.  I'd be grateful if you could take a look.  To test,

git checkout -b markdryan-singlevm-webui master
git pull https://github.com/markdryan/ciao.git singlevm-webui

cd $GOPATH/src/github.com/01org/ciao/testutil/ciao-down
go build
./ciao-down prepare --ui-path <PATH to your ciao-webui repo>

Wait 10 minutes

ciao-down connect

Now you're in the VM.  cd to the ciao-webui directory ( should be the same directory you use on the HOST)

make install-dev

It will fail.  Please help me figure out why.

**Comment by markdryan** [comment]
@jorgevgut I fixed all the proxy and installation issues.  If you follow the instructions above and build with

DESTDIR=/usr/local/nodejs make install-dev

most of the errors are gone.  But I still see some compile errors when building some of the modules,

```
> browserify build/pages/tenant_detail.js -o build/javascripts/bundle_tenant_detail.js


/home/markus/src/ciao-webui/node_modules/d3-scale/index.js:1
export {
^
ParseError: 'import' and 'export' may appear only with 'sourceType: module'

npm ERR! Linux 4.4.0-47-generic
npm ERR! argv "/usr/bin/nodejs" "/usr/bin/npm" "run" "build-detailby-tenant"
npm ERR! node v7.4.0
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE
npm ERR! ciao-webui@1.0.0 build-detailby-tenant: `browserify build/pages/tenant_detail.js -o build/javascripts/bundle_tenant_detail.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ciao-webui@1.0.0 build-detailby-tenant script 'browserify build/pages/tenant_detail.js -o build/javascripts/bundle_tenant_detail.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the ciao-webui package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     browserify build/pages/tenant_detail.js -o build/javascripts/bundle_tenant_detail.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs ciao-webui
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls ciao-webui
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/markus/src/ciao-webui/npm-debug.log
Makefile:49: recipe for target 'install-dev' failed
```

**Comment by markdryan** [comment]
@jorgevgut Note you should update your make file to initial DESTDIR using npm config get prefix

**Comment by markdryan** [comment]
@jorgevgut Okay, the issue seems to be an existing ciao-webui build bug (https://github.com/01org/ciao-webui/issues/129).  What should I do?  Do an

npm install d3-scale

when setting up the VM or will you guys fix this in the web-ui.

**Comment by markdryan** [comment]
@rbradford , @tpepper This should be good to go now.

File: testutil/ciao-down/prepare.go [hunk 1]
```diff
@@ -116,7 +119,30 @@ func prepareSSHKeys(ctx context.Context, ws *workspace) error {
 	return nil
 }
 
+func getProxy(upper, lower string) (string, error) {
```
Comment (tpepper) [inline testutil/ciao-down/prepare.go]: If we keep osprepare, it probably should leverage this as a higher level library utility function.

**Pull request merged**